### PR TITLE
Add enable_unbounded_endpoints core-api flag

### PR DESCRIPTION
**Pull request by malexandru-rdx**
Add `api.core.flags.enable_unbounded_endpoints` config entry (default is `true`, docker env var `RADIXDLT_CORE_API_FLAGS_ENABLE_UNBOUNDED_ENDPOINTS`) which disables all unbounded endpoints (mostly ones that use `dump_component_state` + `/mempool/list`).

### CODE REVIEW

File: docker/config/default.config.envsubst [hunk 1]
```diff
@@ -26,9 +26,11 @@ mempool.reevaluation.max_count=${RADIXDLT_MEMPOOL_REEVALUATION_MAX_COUNT}
 protocol.vertex.max_transaction_count=${RADIXDLT_PROTOCOL_VERTEX_MAX_TRANSACTION_COUNT}
 protocol.vertex.max_total_transactions_size=${RADIXDLT_PROTOCOL_VERTEX_MAX_TOTAL_TRANSACTIONS_SIZE}
 protocol.vertex.max_total_execution_cost_units_consumed=${RADIXDLT_PROTOCOL_VERTEX_MAX_TOTAL_EXECUTION_COST_UNITS_CONSUMED}
+protocol.vertex.max_total_finalization_cost_units_consumed=${RADIXDLT_PROTOCOL_VERTEX_MAX_TOTAL_FINALIZATION_COST_UNITS_CONSUMED}
```
Comment (malexandru-rdx) [inline docker/config/default.config.envsubst:29]: (Note from author)

This is a drive-by fix probably useful for `rcnet-v3`.

**Review by malexandru-rdx — COMMENTED** [review]
After looking at the substates structure for the traversed components, it seems that `/state/validator` is indeed bounded (a big state, but bounded).

File: core-rust/core-api-server/src/core_api/errors.rs [hunk 2]
```diff
@@ -128,6 +128,17 @@ pub(crate) fn assert_matching_network<E: ErrorDetails>(
     Ok(())
 }
 
+pub(crate) fn assert_unbounded_endpoints_flag_enabled<E: ErrorDetails>(
+    state: &CoreApiState,
+) -> Result<(), ResponseError<E>> {
+    if !state.flags.enable_unbounded_endpoints {
+        return Err(client_error(
+            "This endpoint is unbounded and `enable_unbounded_endpoints` flag is disabled.",
```
Comment (dhedey) [inline core-rust/core-api-server/src/core_api/errors.rs:136]: Looks nice, although technically it's the response, not the endpoint itself which is unbounded - so I would slightly rather: "This endpoint is disabled as the response from this endpoint is potentially unbounded, and this node is configured with `enable_unbounded_endpoints` false"

**Review by dhedey — APPROVED** [review]
